### PR TITLE
Update the UI of header and list in Likes Details

### PR DESCRIPTION
**Pull request by jarvislin**
See yWt5gg3nWORhu079Qfv3mS-fi-1135%3A2576

This is a UI redesign in Likes Details. The change scope includes:
1. Header
2. List item

| Post Likes | Comment Likes |
|--|--|
| Screenshot_20240424-110025 | Screenshot_20240424-110051 |

-----

## To Test:

1. Sign in the JP app
3. Switch to the tab of Notifications
4. Find a Post Like notification and click it
5. The header and list should be the same as the design (**without** avatar on the header)
6. Click on the header and the list item, it should keep the original behavior
7. Back to Notifications
8. Find a Comment Like notification and click it
9. The header and list should be the same as the design (**with** avatar on the header)
10. Click on the header and the list item, it should keep the original behavior
11. Done, thank you!

**Known issue**
The comment view reuses the header, it displays incorrectly at the moment. 
I'll fix it when I develop the task of 🤖 Comment Details: Header redesign

-----

## Regression Notes

1. Potential unintended areas of impact

    - notifications

12. What I did to test those areas of impact (or what existing automated tests I relied on)

    - manual

13. What automated tests I added (or what prevented me from doing so)

    - none

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [x] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/engagement/LikedItemViewHolder.kt [hunk 1]
```diff
@@ -19,25 +23,23 @@ class LikedItemViewHolder(
...
-        this.name.text = authorName
         this.snippet.text = likedItem.postOrCommentText
 
         val avatarUrl = WPAvatarUtils.rewriteAvatarUrl(
             likedItem.authorAvatarUrl,
-            rootView.context.resources.getDimensionPixelSize(R.dimen.avatar_sz_small)
+            rootView.context.resources.getDimensionPixelSize(R.dimen.avatar_sz_extra_small)
         )
-
+        avatar.isGone = type == ListScenarioType.LOAD_POST_LIKES
```
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/ui/engagement/LikedItemViewHolder.kt:40]: I noticed that the avatar is also shown in the comment details. I'm not suggesting a change now just noting it down to change this when we work on that screen.

Screenshot_20240424_092023

File: WordPress/src/main/java/org/wordpress/android/ui/engagement/LikedItemViewHolder.kt [hunk 2]
```diff
@@ -72,5 +74,12 @@ class LikedItemViewHolder(
                 likedItem.likedItemPostId
             )
         }
+        val isDarkTheme = itemView.resources.configuration.isDarkTheme()
+        val color = ContextCompat.getColor(
+            itemView.context,
+            if (isDarkTheme) R.color.white_translucent_10
```
Comment (antonis) [inline WordPress/src/main/java/org/wordpress/android/ui/engagement/LikedItemViewHolder.kt]: Nit pick suggestion: Wdyt of moving the dark theme color logic in the xml resource files? E.g. set the `header_root_view.cardBackgroundColor` to a new color resource and differentiate with values in `values/colors.xml` and`values-night/colors.xml`

**Review by antonis — APPROVED** [review]
Awesome work @jarvislin!
The code looks good and the app behaved as expected in my tests on a Pixel 8 Pro/Android 14 🎉 
I left a minor suggestion with https://github.com/wordpress-mobile/WordPress-Android/pull/20700/files#r157736123

File: WordPress/src/main/java/org/wordpress/android/ui/engagement/LikedItemViewHolder.kt (hunk 1 above)
Comment (jarvislin) [reply WordPress/src/main/java/org/wordpress/android/ui/engagement/LikedItemViewHolder.kt:40]: Nice catch! I've updated the visibility of the avatar, but I'll take a deep look at the implementation when I develop the task of 🤖 Comment Details: Header redesign

File: WordPress/src/main/java/org/wordpress/android/ui/engagement/LikedItemViewHolder.kt (hunk 2 above)
Comment (jarvislin) [reply WordPress/src/main/java/org/wordpress/android/ui/engagement/LikedItemViewHolder.kt]: This makes sense to me, I've extracted the background color to `colors.xml`. Thank you for the suggestion!

**Pull request merged**